tests/lint.py

```diff
@@ -0,0 +1,99 @@
+"""Tests performing linter-like checks"""
+
+import ast
+from os.path import basename, dirname, isdir, join
+from os import listdir
+
+
+# Root of the repository. Note that this is relative to this file,
+# so if this file is moved, this may need to be changed:
+source_root = dirname(dirname(__file__))
+
+
+def test_logger_format_strings():
+    """Scan for proper use of logger format strings
+
+    Per @zenhack's comment on issue #629:
+
+    > All over the codebase you can find statments like:
+    >
+    >   logger.error('Foo: %r' % bar)
+    >
+    > The % operator being python's format-string splicing operator. The
+    > problem with this is that the logging functions do the formation
+    > string splicing themselves, i.e. what you want in this case is:
+    >
+    >   logger.error('Foo: %r', bar)
+    >
+    > This opens up the possibility of format-string injection
+    > vulnerabilities. Frankly, this is too easy to do, especially
+    > since in other contexts % is the correct thing. We ought to
+    > (a) make sure all instances of this mistake are fixed, and (b)
+    > come up with a way to catch this mistake automatically going
+    > forward; perhaps some kind of linter.
+
+    This is that linter; it scans the source tree looking for places
+    where the logging functions are called with any first argument
+    that isn't a string literal.
+    """
+
+    def _visit_file(filename):
+        if basename(filename).startswith('.'):
```

Comment:
Arguably the "right" way to do this is by looking at `.gitignore`, but I don't know if I think it's worth the complexity.